q2_metadata/tests/test_random.py

```diff
@@ -194,3 +194,114 @@ def test_shuffle_groups_alt_column_value_prefix(self):
         self.assertEqual(
             set(obs['shuffled.grouping.0'].unique()),
             {'11', '10'})
+
+    def test_shuffle_groups_sample_size_columnid_flag_no_input(self):
+        md = qiime2.CategoricalMetadataColumn(
+            pd.Series(['a', 'b', 'a', 'b'], name='groups',
+                      index=pd.Index(['sample1', 'sample2', 'sample3', 's4'],
+                                     name='id'))
+        )
+        # expected number of rows and columns in result
+        obs = shuffle_groups(md, n_columns=1,
+                             column_name_prefix='shuffled.grouping.',
+                             column_value_prefix='fake.group.',
+                             )
+        self.assertEqual(obs.shape, (4, 1))
+
+        # expected column names (the original should not be in the result)
+        self.assertFalse('groups' in obs.columns)
+        self.assertTrue('shuffled.grouping.0' in obs.columns)
+
+        # correct number of groups in the new column
+        self.assertEqual(len(obs['shuffled.grouping.0'].unique()), 2)
+
+        # correct group names in new column
+        self.assertEqual(
+            set(obs['shuffled.grouping.0'].unique()),
+            {'fake.group.1', 'fake.group.0'})
+
+    def test_shuffle_groups_sample_size_columnid_flag_true(self):
+        md = qiime2.CategoricalMetadataColumn(
+            pd.Series(['a', 'b', 'a', 'b'], name='groups',
+                      index=pd.Index(['sample1', 'sample2', 'sample3', 's4'],
+                                     name='id'))
+        )
+
+        # expected number of rows and columns in result
+        obs = shuffle_groups(md, n_columns=1,
+                             column_name_prefix='shuffled.grouping.',
+                             column_value_prefix='fake.group.',
+                             encode_sample_size=True)
+        self.assertEqual(obs.shape, (4, 1))
+
+        # expected column names (the original should not be in the result)
+        self.assertFalse('groups' in obs.columns)
+        self.assertTrue('shuffled.grouping.0' in obs.columns)
+
+        # correct number of groups in the new column
+        self.assertEqual(len(obs['shuffled.grouping.0'].unique()), 2)
```

Comment:
```suggestion
```
Similar suggestion as the last test - I'd remove the extra assertions that aren't testing specifically what you're trying to prove with this individual test.